fedimint-core/src/module/mod.rs

```diff
@@ -584,29 +597,220 @@ pub struct ApiVersion {
     pub minor: u32,
 }
 
+/// Multiple, disjoin, minimum required or maximum supported, api versions.
+///
+/// If a given component can (potentially) support multiple different (distinct
+/// major number), of an API, this type is used to express it.
+///
+/// Each element must have a distinct major api number, and means
+/// either minimum required API version of this major number, or maximum
+/// supported version of this major number.
```

Comment:
Who does the interpretation of this struct? The client?

Reply:
Whatever code using it. But the client common api versions discovery does happen on a client side.